### PR TITLE
Add notation section & remove footnote remarks in favor of it

### PR DESCRIPTION
**Pull request by WhatisRT**
# Description

Lots of notations still need to be explained (or removed), but this is a start.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] Code is formatted according to [CONTRIBUTING.md](https://github.com/input-output-hk/formal-ledger-specifications/blob/master/CONTRIBUTING.md)
- [x] Self-reviewed the diff

### CODE REVIEW

File: src/Ledger/Notation.lagda [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+In this section, we introduce some notations used in this document.
+
+\begin{description}
+\item[Propositions, sets and types] This document loosly treats sets and types as the same thing.
+\item[Sums and products] The sum of \AgdaBound{A} and \AgdaBound{B} (also called their disjoint
+  union, coproduct, etc.) is denoted by \AgdaBound{A} \coproduct \AgdaBound{B}, and their product
+  is denoted by \AgdaBound{A} \agdatimes \AgdaBound{B}.
+\item[Record types] Record types are explained in Appendix \ref{sec:appendix-agda-essentials}.
+\item[Postfix projections] TODO, if we want them in the PDF
```
Comment (omelkonian) [inline src/Ledger/Notation.lagda:16]: Add a normal sentence for postfix projections (e.g. "Record fields are simply projection functions from a value of the record to said field and can be conveniently written in postfix form; for instance, getting the e-mail `address` of a `user` would be denoted `user .address`"); we can then remove it if we decide no such projections are ever rendered on PDF.

**Review by omelkonian — CHANGES_REQUESTED** [review]

File: src/Ledger/Notation.lagda [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+module Ledger.Notation where
+\end{code}
+
+In this section, we introduce some notations used in this document.
+
+\begin{description}
+\item[Propositions, sets and types] This document loosly treats sets and types as the same thing.
+\item[Sums and products] The sum of \AgdaBound{A} and \AgdaBound{B} (also called their disjoint
+  union, coproduct, etc.) is denoted by \AgdaBound{A} \coproduct \AgdaBound{B}, and their product
+  is denoted by \AgdaBound{A} \agdatimes \AgdaBound{B}.
```
Comment (omelkonian) [inline src/Ledger/Notation.lagda:15]: I would rephrase this as "The product of  \AgdaBound{A} and \AgdaBound{B}  is denoted by \AgdaBound{A} \agdatimes \AgdaBound{B}; their sum (a.k.a. disjoint union or coproduct) is denoted \AgdaBound{A} \coproduct \AgdaBound{B}."

File: src/Ledger/Notation.lagda [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+\item[Record types] Record types are explained in Appendix \ref{sec:appendix-agda-essentials}.
```
Comment (omelkonian) [inline src/Ledger/Notation.lagda:16]: ...in Appendix~\ref...

(`~` should generally be used consistently  with `\ref`)

**Review by omelkonian — APPROVED** [review]
Some small changes, otherwise LGTM.

File: src/Ledger/Notation.lagda (hunk 2 above)
Comment (WhatisRT) [reply src/Ledger/Notation.lagda]: I prefer the original order, since that's the order of the header as well. But I did slightly rephrase it.
